lean4-info.el

```diff
@@ -82,19 +82,26 @@
 (lsp-defun lean4-diagnostic-full-end-line ((&lean:Diagnostic :full-range (&Range :end (&Position :line))))
   line)
 
-(defun lean4-mk-message-section (caption errors)
+(defun lean4-mk-message-section (uri caption errors)
   (when errors
     (magit-insert-section (magit-section)
       (magit-insert-heading caption)
       (magit-insert-section-body
         (dolist (e errors)
           (-let (((&Diagnostic :message :range (&Range :start (&Position :line :character))) e))
-            (magit-insert-section (magit-section)
-              (magit-insert-heading (format "%d:%d: " (1+ (lsp-translate-line line)) (lsp-translate-column character)))
+	    (magit-insert-section (magit-section)
+	      (magit-insert-heading)
+	      (insert-button (format "%s %d:%d" uri (1+ (lsp-translate-line line)) (lsp-translate-column character))
```

Comment:
I should style the button as if it were a magit header. It currently looks like a raw link. I haven't figured this out yet.

Reply:
I think the intended way is to modify the section's keymap, e.g. how `RET` in magit goes to whatever is underneath the cursor https://github.com/magit/magit/blob/4e29d5827cb77a7fbcff57033a099e23b8edd424/lisp/magit-status.el#L538-L540